lms/product/__init__.py

```diff
@@ -13,6 +12,9 @@ def includeme(config):
     config.include("lms.product.canvas")
     config.include("lms.product.d2l")
 
+    # Lazy import to avoid circular dependencies
+    from lms.product.factory import get_product_from_request
```

Comment:
This triggers an import of each product and their plugins so its quite heavy.

I'm sure shuffling things around might be a better solution but it will make this quite a bigger PR.